File: Easy/Count Asterisks/CountAsterisks.cpp
// https://leetcode.com/problems/count-asterisks

class Solution {
public:
  int countAsterisks(string s) {
    ios::sync_with_stdio(0), cin.tie(0), cin.tie(0);
    int sep = 0, count = 0;
    for (int i = 0; i < s.size(); i++) {
      if (s[i] == '|') {
        sep++;
      }
      if (!(sep & 1) && s[i] == '*') {
        count++;
      }
    }
    return count;
  }
};

File: Easy/Count Asterisks/CountAsterisks.js
// https://leetcode.com/problems/count-asterisks

/**
 * @param {string} s
 * @return {number}
 */
var countAsterisks = function (s) {
  let sep = 0,
    count = 0;
  for (let i = 0; i < s.length; i++) {
    if (s[i] == "|") {
      sep++;
    }
    if (sep % 2 == 0 && s[i] == "*") {
      count++;
    }
  }
  return count;
};
